### PR TITLE
respect cert TTL in client side for k8s RA

### PR DESCRIPTION
**Pull request by irisdingbj**
**Please provide a description of this PR:**
user can specify the workload cert TTL via `SECRET_TTL`, however this option is ignored if we use K8s RA as cert provider. 
cert-manager supports to use `experimental.cert-manager.io/request-duration` annotation in  k8s CSR, So this PR will add the annotation to the CSR istio generated to support the eco-system.

### CODE REVIEW

**Comment by shankgan** [comment]
Are we sure the Kubernetes CSR API gives us no mechanism to specify the cert TTL (I wasn't able to find the relevant API as well). We can use certManager annotations if there is no other canonical way to configure this.

File: security/pkg/k8s/chiron/utils.go [hunk 1]
```diff
@@ -211,6 +213,9 @@ func submitCSR(clientset clientset.Interface,
 					SignerName: signerName,
 				},
 			}
+			if requestedLifetime.String() != "0s" {
```
Comment (howardjohn) [inline security/pkg/k8s/chiron/utils.go:216]:
```suggestion
			if requestedLifetime != time.Duration(0) {
```

**Comment by irisdingbj** [comment]
> Are we sure the Kubernetes CSR API gives us no mechanism to specify the cert TTL (I wasn't able to find the relevant API as well). We can use certManager annotations if there is no other canonical way to configure this.

Yes, the CSR API itself has no way to sepcify the cert TTL this is also the reason why cert-manager is using annotation in the CSR to support this.

File: security/pkg/k8s/chiron/controller.go [hunk 2]
```diff
@@ -205,8 +205,9 @@ func (wc *WebhookController) upsertSecret(secretName, dnsName, secretNamespace s
 		return nil
 	}
 
+	requestedLifetime := time.Duration(0)
```
Comment (hzxuzhonghu) [inline security/pkg/k8s/chiron/controller.go:208]: What does 0 mean?



It seems not quite right https://github.com/jetstack/cert-manager/blob/75d91bcb297d8a1506abd376ea378cf760c4f7f1/pkg/util/pki/csr.go#L380

Comment (irisdingbj) [reply security/pkg/k8s/chiron/controller.go:208]: This is for Backward compatiblity : we will use  the `kubernetes.io/legacy-unknow' by default. In that case, no TTL can be specified.  So using a  '0' means we do not want to add related annotation to the CSR.  See : https://github.com/istio/istio/pull/34484/files#diff-6fa317e12eadb34d01780d845992c68fb30c869c9cc9127fd6458e728dacec1fR247

Comment (hzxuzhonghu) [reply security/pkg/k8s/chiron/controller.go:208]: make sense

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by irisdingbj** [comment]
@shankgan @howardjohn PTAL, Thanks!

File: pkg/config/constants/constants.go [hunk 3]
```diff
@@ -125,4 +125,6 @@ const (
 	// CertProviderNone does not create any certificates for the control plane. It is assumed that some external
 	// load balancer, such as an Istio Gateway, is terminating the TLS.
 	CertProviderNone = "none"
+
+	RequestLifeTimeAnnotationForCertManager = "experimental.cert-manager.io/request-duration"
```
Comment (shankgan) [inline pkg/config/constants/constants.go:129]: Please add comment here to explain why we are using CertManager Annotation. Also, could we move this constant somewhere inside the chiron package?

**Review by shankgan — APPROVED** [review]

File: pkg/config/constants/constants.go (hunk 3 above)
Comment (irisdingbj) [reply pkg/config/constants/constants.go]: done

**Comment by irisdingbj** [comment]
@howardjohn PTAL, Thanks!

**Review by howardjohn — APPROVED** [review]
approve t&r change

**Pull request merged**